docs/architecture/adr/0016-web-push-sunset-signalr.md

```diff
@@ -0,0 +1,109 @@
+Bitwarden-hosted version is backed by dozens of virtual machines each supporting tens of thousands
+of concurrent connections. With the move to less manual component maintenance in the cloud --
+specifically Kubernetes -- the burden of keeping up the large amount of connections with
+[pods][podlife] is complex and performance spikes are expected.
+
+### WebSockets
+
+Usage of SignalR, even with much larger workloads, has at face value worked well, but the nature of
+the work being performed and usage of persistent WebSockets connections for certain clients has
+become tenuous. In aggregate the amount of input and output from the notifications service is quite
+large and is used almost entirely for background-oriented operations. Various new technologies have
+come out that are better suited towards synchronization work. The technology or protocol used must
+be usable at large scale in the cloud as well as for self-hosted installations.
+
+Browser extension changes and mandates such as [Manifest V3][mv3] additionally present support
+problems with long-lived background connections.
+
+### Cost and Reliability
+
+New solutions must be able to not only scale to essentially infinite connections but balance cost
+with user growth, all the while delivering high availability. Existing options for mobile
+notifications specifically either have device limits (e.g. [Azure Notification Hubs][hubspricing])
+or lack service level guarantees while not offering modern technologies. Furthermore, some clients
+(e.g. [F-Droid][fdroid]) cannot utilize well-established (albeit proprietary) push backends. A
+single service provider is desired for as much functionality as possible, while still offering
+flexibility for self-host.
+
+## Considered Options
+
+- **Keep the SignalR solution and continue to scale up** - Maintain the cluster of notifications
+  service virtual machines and keep pushing for a larger set to handle scale. Move all mobile
+  notifications to the custom solution and abandon Azure Notification Hubs.
+- **Work around Azure Notification Hubs limits** - Since all mobile devices, even for self-hosted
+  installations, must utilize the Bitwarden cloud for push due to certificate security, devise a
+  solution that shards devices across many Notification Hubs within new subscriptions.
+- **Adopt a new offering for mobile push notifications only** - Use something other than Azure
+  Notification Hubs that doesn't have limits, perhaps with some technology sacrifices.
+- **Adopt a new offering for non-mobile push notifications only** - Use something other than SignalR
+  like a homegrown [Web Push][webpush] backend inside the notifications service.
+- **Adopt a new combined push service provider and utilize Web Push** - Not only migrate away from
+  Azure Notification Hubs but also utilize Web Push with a single, reliable service provider. Host a
+  compatible Web Push backend for self-host with the clients that need it.
```

Comment:
Can you expand on hosting a compatible web push backend, does this mean extending our relay notifications to web push or try to find a way for self host to push directly to browser push services?